internal/topology/variables/cluster_variable_validation_test.go

```diff
@@ -900,6 +900,223 @@ func Test_ValidateClusterVariable(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "Valid object with x-kubernetes-preserve-unknown-fields",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name:     "testObject",
+				Required: true,
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
+						Properties: map[string]clusterv1.JSONSchemaProps{
+							"requiredProperty": {
```

Comment:
q: is this required in the sense of "a value must be provided" or just a "know property"

Reply:
Just copy & paste from somewhere above. Replaced with something neutral

Reply:
It seems it is not resolved (or at least not in all test cases
what about
s/requiredProperty/knowProperty
s/boolProperty/unknowProperty

Reply:
The problem is that the property is not always known. In some cases we just use the preserve unknown thing

Reply:
Hm. I'll change it to that, still makes sense even if we are using preserve. Missed that :)

Reply:
Not sure what I did with the last change - probably too much multi-tasking

Reply:
Should be fixed now